aos_cd_jobs/updater.py

```diff
@@ -20,25 +18,42 @@ def list_jobs(repo):
     return jobs
 
 def create_remote_branch(repo, name):
-    initialize_orphan_branch(repo, name)
-    populate_branch(repo, name)
+    branch = checkout_branch(repo, name)
+    populate_branch(repo, branch)
     publish_branch(repo, name)
 
-def initialize_orphan_branch(repo, name):
-    repo.heads.master.checkout(orphan=name)
-
-def populate_branch(repo, name):
-    directory = join(repo.working_dir, JOBS_DIRECTORY, name)
-    create_job_file_tree(repo.working_dir, directory)
+def checkout_branch(repo, name):
+    branch = getattr(repo.branches, name, None)
+    if branch is None:
```

Comment:
I'd suggest to make the logic more explicit instead of having the fallback, create branch if it doesn't exist before calling into this one